### ISSUE TITLE
`bin/plugin make` fails on Windows

### ISSUE THREAD

**Issue by firephreek** [flutter/flutter-intellij#6052]
## Steps to Reproduce
- On a Windows operating system, run `bin/plugin` or `dart tool/plugin/bin/main.dart make`

## Result
The script uses OS specific calls and is not able to perform the necessary operations on Windows as it does on Linux.

## Version info
Info attached: [flutter_doctor.txt](https://github.com/flutter/flutter-intellij/files/8337139/flutter_doctor.txt)

**Comment by stevemessick** [comment]
The unfortunate truth is that no one on the Flutter plugin team has any deep experience with Windows.

We'd be glad to review PRs to make the plugin tool work on Windows, but we're not likely to do it ourselves.

FWIW, I just checked out the setup instructions for Windows a couple weeks ago and found that it is still mostly accurate. If you got to the point of trying to build the plugin, did you also get IntelliJ set up to work with the sources for IntelliJ and the two plugins (Dart, Flutter)?

**Comment by firephreek** [comment]
I'm ex-G, I totally understand. Already put in a PR for the work after
dealing with some CLA issues.

I only haven't gotten me env setup because of an IJ version mismatch with
the plugin. Should be straightforward to deal with though.

**Comment by Hixie** [comment]
There was a PR written by @firephreek to solve this issue (see link below) but unfortunately it was not completed so the PR has been closed. If anyone would like to start from that PR and apply the review comments and drive this PR to completion, you are definitely welcome to do so!

PR: https://github.com/flutter/flutter-intellij/pull/6054

**Comment by stevemessick** [comment]
Thanks @Hixie and @firephreek. Also note that if you install git for Windows the code does work as-is. This issue is about using Windows in a more native style.

**Comment by firephreek** [comment]
@Hixie @stevemessick Sorry about that guys, got pulled away by life.  Finally managed to circle back on it and I'm putting together a new PR for this issue, just running into a few issues re: Gradle and the `compileClasspath` target. It's likely due to my own lack of deep Gradle familiarity, but there's some weird caching behavior that happens where the 'compileClasspath' dependency is never 'refreshed'.

Ex: In early testing on Windows 10, the linux.targ.gz library is downloaded and unpacked to the artifact folder. After realizing I'd need the windows binary, I deleted the `flutter-intellij/artifacts` path and unpacked the Windows versions of IdeaIC and Android-Studio, but would still get Gradle exceptions during the `compileKotlin` phase indicating that it was trying to compile `lib/ffmpeg...-linux.jar` libraries instead of the windows versions that were now in the artifact folder.  I ended up having to delete the entire gradle cache in my user directory, rebuilt from scratch, but now it links to the Android Studio libraries.  So whenever I execute the `[runIdea]` target, it launches the latest AS, not IntelliJ.  

Again, this is likely to me not having a deep familiarity with Gradle. Where should I start the troubleshooting on this one? I don't want to push this PR and cause noise until I have a better grasp of what's happening...

[Branch under development](https://github.com/firephreek/flutter-intellij/tree/support_windows_platform)
